From b92167f8a027dd4c02ad86626009097eb53712d2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Sun, 7 Jul 2013 12:06:31 +0200
Subject: [PATCH 013/127] wined3d: Preload buffers if streamsrc is not dirty
Reply-To: wine-devel <wine-devel@winehq.org>

FIXME: Merge this with the preload calls done by
context_update_stream_info, preload only used buffers

FIXME2: Merge this patch with the patch that removes
PreLoad from buffer_unmap
---
 dlls/wined3d/context.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/dlls/wined3d/context.c b/dlls/wined3d/context.c
index 11d86c5..864a0c7 100644
--- a/dlls/wined3d/context.c
+++ b/dlls/wined3d/context.c
@@ -2872,6 +2872,15 @@ BOOL context_apply_draw_state(struct wined3d_context *context, struct wined3d_de
     context_preload_textures(context, state);
     if (isStateDirty(context, STATE_VDECL) || isStateDirty(context, STATE_STREAMSRC))
         context_update_stream_info(context, state);
+    else
+    {
+        for (i = 0; i < sizeof(state->streams) / sizeof(*state->streams); i++)
+        {
+            if (state->streams[i].buffer)
+                buffer_internal_preload(state->streams[i].buffer, context, state);
+        }
+    }
+
     if (state->index_buffer)
     {
         if (context->stream_info.all_vbo)
-- 
1.8.1.5

